src/js/components/TaskFileDownloadComponent.jsx

```diff
@@ -0,0 +1,98 @@
+var React = require("react/addons");
+var classNames = require("classnames");
+
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+var MesosStore = require("../stores/MesosStore");
+
+function matchFileName(name) {
+  return (file) => file.name === name;
+}
+
+var TaskFileLinkComponent = React.createClass({
+  displayName: "TaskFileLinkComponent",
+  propTypes: {
+    children: React.PropTypes.node,
+    className: React.PropTypes.string,
+    fileName: React.PropTypes.string.isRequired,
+    task: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      file: this.getFile(),
+      requested: false
+    };
+  },
+
+  componentWillMount: function () {
+    MesosStore.on(MesosEvents.CHANGE, this.onMesosChange);
+  },
+
+  componentWillUnmount: function () {
+    MesosStore.removeListener(MesosEvents.CHANGE, this.onMesosChange);
+  },
+
+  getFile: function () {
+    var props = this.props;
+    var task = this.props.task;
+    var taskId = task.id || task.taskId;
+    var files = MesosStore.getTaskFiles(taskId);
+    if (files) {
+      return files.filter(matchFileName(props.fileName))[0];
+    }
+    return null;
+  },
+
+  onMesosChange: function () {
+    var file = this.getFile();
+    var requested = this.state.requested;
+    // Start download if file was requested by the user
+    if (file && requested) {
+      window.open(file.download);
+      requested = false;
```

Comment:
As the change event is fired quite often, we can't update the request state unless we have the correct file, thus we can't simply remove the line. Besides that, it's a primitive, thus it's copied and the state isn't mutated.